client/state/preferences/actions.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import {
+	PREFERENCES_SET,
+	PREFERENCES_RECEIVE,
+	PREFERENCES_FETCH,
+	PREFERENCES_FETCH_SUCCESS,
+	PREFERENCES_FETCH_FAILURE
+} from 'state/action-types';
+import { USER_SETTING_KEY } from './constants';
+
+export function fetchPreferences() {
+	return ( dispatch ) => {
+		dispatch( { type: PREFERENCES_FETCH } );
+		return wpcom.me().settings().get()
+		.then( data => dispatch( {
+			type: PREFERENCES_FETCH_SUCCESS,
+			data
+		} ) )
+		.catch( ( data, error ) => dispatch( {
+			type: PREFERENCES_FETCH_FAILURE,
+			data,
+			error
+		} ) );
+	};
+}
+
+export const setPreference = ( key, value ) => ( dispatch, getState ) => {
+	dispatch( {
+		type: PREFERENCES_SET,
+		key,
+		value
+	} );
+	const settings = {};
+	settings[ USER_SETTING_KEY ] = getState().preferences.values;
```

Comment:
Could we maybe add a comment here to let folks know that this is a special case? We generally avoid fetching state in actions, but it's unavoidable in this case.